crates/dids/src/method/dht/mod.rs

```diff
@@ -0,0 +1,126 @@
+mod bep44;
+mod convert;
+pub mod registered_types;
+
+use std::sync::Arc;
+
+use crypto::Curve;
+use keys::key_manager::KeyManager;
+use reqwest::blocking::Client;
+
+use crate::{
+    bearer::BearerDid,
+    document::{Document, Service, VerificationMethod},
+    identifier::Identifier,
+};
+
+use self::{
+    bep44::encode_bep44_message, convert::document_packet::document_to_packet,
+    registered_types::RegisteredDidType,
+};
+
+use super::{Create, MethodError};
+
+const DEFAULT_GATEWAY_URL: &str = "https://diddht.tbddev.org";
+
+/// Concrete implementation for a did:dht DID
+pub struct DidDht;
+
+/// Options that can be used to create a did:dht DID
+pub struct DidDhtCreateOptions {
+    pub publish: bool,
+    // TODO: gatewayUri - default 'https://diddht.tbddev.org'
+    pub also_known_as: Option<Vec<String>>,
+    pub controller: Option<Vec<String>>,
+    pub service: Option<Vec<Service>>,
+    pub registered_type: Option<Vec<RegisteredDidType>>,
+    pub verification_methods: Option<Vec<VerificationMethod>>,
+}
+
+impl Create<DidDhtCreateOptions> for DidDht {
+    fn create(
+        key_manager: Arc<dyn KeyManager>,
+        options: DidDhtCreateOptions,
+    ) -> Result<BearerDid, MethodError> {
+        // Generate private and public keypair
+        let key_alias = key_manager.generate_private_key(
+            Curve::Ed25519,
+            Some("0".to_string()),
+        )?;
+        let public_key = key_manager.get_public_key(&key_alias)?;
+
+        // Create did uri
+        let public_jwk = public_key.jwk()?;
+        let jwk_string = serde_json::to_string(public_jwk.as_ref()).map_err(|_| {
+            MethodError::DidCreationFailure("failed to serialize public jwk".to_string())
+        })?;
+        let suffix = zbase32::encode_full_bytes(jwk_string.as_bytes());
+        let uri = format!("did:dht:{}", suffix);
+        let identifier = Identifier::parse(&uri).map_err(|e| {
+            MethodError::DidCreationFailure(format!("Failed to parse did:jwk uri {} {}", &uri, e))
```

Comment:
```suggestion
            MethodError::DidCreationFailure(format!("Failed to parse did:dht uri {} {}", &uri, e))
```

on a similar note, you could automatically map the error by adding this to the `IdentifierError`

```rust
    #[error(transparent)]
    IdentifierError(#[from] IdentifierError),
```

also if you're curious, go check out `jws/src/lib.rs` and you'll see this for mapping serde serialization errors (a few lines above this line)

```rust
impl From<SerdeJsonError> for JwsError {
    fn from(err: SerdeJsonError) -> Self {
        JwsError::SerdeJsonError(err.to_string())
    }
}
```